addons/l10n_rs_edi/views/account_move.xml

```diff
@@ -0,0 +1,52 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+    <record id="view_move_form_inherit" model="ir.ui.view">
+        <field name="name">account.move.form.inherit</field>
+        <field name="model">account.move</field>
+        <field name="inherit_id" ref="account.view_move_form"/>
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='delivery_date']" position="after">
+                <field name="l10n_rs_tax_date_obligations_code" 
+                    invisible="country_code != 'RS' and move_type != 'out_invoice'"
+                    required="country_code == 'RS' and move_type == 'out_invoice'"/>
+            </xpath>
+
+            <xpath expr="//field[@name='delivery_date']" position="attributes">
+                <attribute name="required">l10n_rs_tax_date_obligations_code == '35'</attribute>
```

Comment:
You can't do that. You are modifying the required for all companies, not only the RS ones. Add a condition with the country code